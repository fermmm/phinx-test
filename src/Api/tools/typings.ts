export interface CharactersResponse {
   code: number;
   status: string;
   copyright: string;
   attributionText: string;
   attributionHTML: string;
   data: {
      offset: number;
      limit: number;
      total: number;
      count: number;
      results: CharacterData[];
   };
   etag: string;
}

export interface CharactersComicsResponse {
   code: number;
   status: string;
   copyright: string;
   attributionText: string;
   attributionHTML: string;
   data: {
      offset: number;
      limit: number;
      total: number;
      count: number;
      results: ComicData[];
   };
   etag: string;
}

export interface ComicData {
   id: number;
   digitalId: number;
   title: string;
   issueNumber: number;
   variantDescription: string;
   description: string;
   modified: "Date";
   isbn: string;
   upc: string;
   diamondCode: string;
   ean: string;
   issn: string;
   format: string;
   pageCount: number;
   textObjects: [
      {
         type: string;
         language: string;
         text: string;
      },
   ];
   resourceURI: string;
   urls: [
      {
         type: string;
         url: string;
      },
   ];
   series: {
      resourceURI: string;
      name: string;
   };
   variants: [
      {
         resourceURI: string;
         name: string;
      },
   ];
   collections: [
      {
         resourceURI: string;
         name: string;
      },
   ];
   collectedIssues: [
      {
         resourceURI: string;
         name: string;
      },
   ];
   dates: [
      {
         type: string;
         date: "Date";
      },
   ];
   prices: [
      {
         type: string;
         price: "float";
      },
   ];
   thumbnail: ThumbnailData;
   images: [
      {
         path: string;
         extension: string;
      },
   ];
   creators: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
            role: string;
         },
      ];
   };
   characters: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
            role: string;
         },
      ];
   };
   stories: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
            type: string;
         },
      ];
   };
   events: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
         },
      ];
   };
}

export interface CharacterData {
   id: number;
   name: string;
   description: string;
   modified: "Date";
   resourceURI: string;
   urls: [
      {
         type: string;
         url: string;
      },
   ];
   thumbnail: ThumbnailData;
   comics: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
         },
      ];
   };
   stories: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
            type: string;
         },
      ];
   };
   events: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
         },
      ];
   };
   series: {
      available: number;
      returned: number;
      collectionURI: string;
      items: [
         {
            resourceURI: string;
            name: string;
         },
      ];
   };
}

export interface ThumbnailData {
   path: string;
   extension: string;
}
