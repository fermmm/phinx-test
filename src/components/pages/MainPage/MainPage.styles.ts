// tslint:disable: typedef
import styled from "styled-components";

export const MainPageContainer = styled("div")`
   background-color: ${props => props.theme.colors.background1};
   width: 100vw;
   height: 100vh;
`;
