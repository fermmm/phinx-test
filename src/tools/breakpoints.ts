// tslint:disable: typedef
import { css, CSSObject, SimpleInterpolation } from "styled-components";

export const breakpoints: BreakPoints = {
   xs: "480px",
   sm: "768px",
   md: "992px",
   lg: "1200px",
};

export interface BreakPoints {
   xs: string;
   sm: string;
   md: string;
   lg: string;
}

export const minWidth = Object.keys(breakpoints).reduce((accumulator, label) => {
   accumulator[label] = (
      first: TemplateStringsArray | CSSObject,
      ...interpolations: SimpleInterpolation[]
   ) => css`
      @media (min-width: ${breakpoints[label]}) {
         ${css(first, ...interpolations)};
      }
   `;
   return accumulator;
}, {}) as BreakpointsQueries;

export const maxWidth = Object.keys(breakpoints).reduce((accumulator, label) => {
   accumulator[label] = (
      first: TemplateStringsArray | CSSObject,
      ...interpolations: SimpleInterpolation[]
   ) => css`
      @media (max-width: ${breakpoints[label]}) {
         ${css(first, ...interpolations)};
      }
   `;
   return accumulator;
}, {}) as BreakpointsQueries;

type BreakpointsQueries = Record<
   keyof BreakPoints,
   (first: TemplateStringsArray | CSSObject, ...interpolations: SimpleInterpolation[]) => string
>;
