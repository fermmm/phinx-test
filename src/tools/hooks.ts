import qs from "qs";
import { useHistory } from "react-router-dom";

export interface UrlQuery {
   character?: string;
   comic?: string;
}

export function useUrlQuery(): UrlQuery {
   return qs.parse(useHistory().location.search, { ignoreQueryPrefix: true });
}