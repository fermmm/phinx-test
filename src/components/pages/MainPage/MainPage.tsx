import React, { FC, useEffect } from "react";
import { hot } from "react-hot-loader";
import { Route, Switch } from "react-router";
import {
   clearCharactersListAction,
   randomCharacterRequestAction,
   requestCharactersAction,
} from "../../../context/actions/characters/charactersActions";
import { comicFilterAction } from "../../../context/actions/comics/comicsActions";
import { GlobalStateContext } from "../../../context/store";
import { UrlQuery, useUrlQuery } from "../../../tools/hooks";
import CharacterGallery from "./CharacterGallery/CharacterGallery";
import ComicFullInfo from "./ComicFullInfo/ComicFullInfo";
import { MainPageContainer } from "./MainPage.styles";
import SearchBar from "./SearchBar/SearchBar";

const MainPage: FC = () => {
   const urlQuery: UrlQuery = useUrlQuery();
   const { state, dispatch } = React.useContext(GlobalStateContext);

   useEffect(() => {
      requestDataUsingUrlQuery();
   }, []);

   const requestDataUsingUrlQuery = async () => {
      dispatch(clearCharactersListAction());
      if (urlQuery.character) {
         dispatch(requestCharactersAction(urlQuery.character));

         if (urlQuery.comic) {
            dispatch(comicFilterAction(urlQuery.comic));
         }
      } else {
         dispatch(randomCharacterRequestAction());
      }
   };

   return (
      <MainPageContainer>
         <SearchBar />
         <Switch>
            <Route exact path={"/"} component={CharacterGallery} />
            <Route path={"/comic"} component={ComicFullInfo} />
         </Switch>
      </MainPageContainer>
   );
};

export default hot(module)(MainPage);
