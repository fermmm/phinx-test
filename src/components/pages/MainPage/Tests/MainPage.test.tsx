import { shallow } from "enzyme";
import React from "react";
import MainPage from "../MainPage";
import * as hooks from "../../../../tools/hooks";

jest.spyOn(hooks, 'useUrlQuery').mockImplementation(() => ({}));

describe("<MainPage />", () => {
   test("Renders correctly", () => {
      expect(shallow(<MainPage />)).toMatchSnapshot();
   });
});
