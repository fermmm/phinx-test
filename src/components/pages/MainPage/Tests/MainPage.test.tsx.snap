// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> Renders correctly 1`] = `
<MainPagestyles__MainPageContainer>
  <SearchBar />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/comic"
    />
  </Switch>
</MainPagestyles__MainPageContainer>
`;
